src/ceph-disk/ceph_disk/main.py

```diff
@@ -2773,11 +2752,11 @@ def populate_data_path(self, path, *to_prepare_list):
         else:
             LOG.debug('Preparing osd data dir %s', path)
 
-        if self.args.osd_uuid is None:
-            self.args.osd_uuid = str(uuid.uuid4())
+        assert self.args.osd_uuid is not None
 
         write_one_line(path, 'ceph_fsid', self.args.cluster_uuid)
         write_one_line(path, 'fsid', self.args.osd_uuid)
+        write_one_line(path, 'whoami', self.????.osd_id)
```

Comment:
i didn't know how/where to store the state in the Prepare class so that later stages can get at it.